himportscan/src/HImportScan/ImportScanner.hs

```diff
@@ -55,9 +60,15 @@ instance Aeson.ToJSON ScannedImports where
       , ("usesTH", Aeson.toJSON usesTH)
       ]
 
+-- Since it is easier to store in JSON format,
+-- the type `ImportMethod`, isomorphic to `Bool`
+-- is casted during the translation to JSON.
 instance Aeson.ToJSON ModuleImport where
-  toJSON (ModuleImport maybePackageName moduleName) =
-    Aeson.toJSON $ catMaybes [maybePackageName, Just moduleName]
+  toJSON (ModuleImport importMethod maybePackageName moduleName) =
+    Aeson.object
+      [ ("isSourceImported", Aeson.Bool (importMethod == SourceImport))
+      , ("moduleName", Aeson.toJSON $ catMaybes [maybePackageName, Just moduleName])
+      ]
```

Comment:
Since you're already using an object here anyways, I think it would be an overall improvement to do the same for `moduleName` as well and make this a flat object with three keys. It also simplifies `UnmarshalJSON`, I think.